### PR TITLE
issue at_recv memory leak problem

### PR DESCRIPTION
**Pull request by seblee**
## 拉取/合并请求描述：(PR description)

[
 at_recv_notice_cb() 函数中

    if (sock == RT_NULL)
    {
          return ;
    }
改为

    if (sock == RT_NULL)
    {
        rt_free((char *)buff);
        return ;
    }

  环境为 stm32f407 + sim7600 通信中出现的问题
]

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [*] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [*] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [*] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other style
- [*] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [*] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [*] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [*] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality

### CODE REVIEW

**Comment by armink** [comment]
提交 pr 前请先将本地更新至最新代码

**Comment by seblee** [comment]
@armink 
更新到最新后程序不运行了 

请看
https://github.com/RT-Thread/rt-thread/issues/2572

**Comment by BernardXiong** [comment]
请仔细阅读PR消息模板里的内容，后面的message内容不应该被修改，而是直接在浏览器上一一对比更改后，在浏览器中挨个打狗

File: components/net/at/at_socket/at_socket.c [hunk 1]
```diff
@@ -496,8 +496,10 @@ static void at_recv_notice_cb(int socket, at_socket_evt_t event, const char *buf
 
     sock = at_get_socket(socket);
     if (sock == RT_NULL)
+    {
+        rt_free((char *)buff);
```
Comment (armink) [inline components/net/at/at_socket/at_socket.c]: 只更新这一段就行了

**Pull request closed without merging**